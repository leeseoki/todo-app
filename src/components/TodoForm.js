import React, { Component } from 'react';

class TodoForm extends Component {
  constructor(props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this)
  }

  onSubmit(event) {
    event.preventDefault()
    var itemValue = this.refs.item.value
    if(itemValue) {
      this.props.addItem(itemValue)
      this.refs.form.reset()
    }
  }

  render() {
    return (
      <form ref="form" onSubmit={this.onSubmit} className="form-inline">
        <input ref="item" type="text" className="form-control"  placeholder="Enter new todo" />
        <button type="submit" className="btn btn-primary">Add</button>
      </form>
    )
  }
}

export default TodoForm

