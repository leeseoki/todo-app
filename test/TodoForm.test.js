import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import TodoForm from '../src/components/TodoForm'

describe('TodoForm Component', () => {

  beforeAll(() => {
      Enzyme.configure({ adapter: new Adapter() })
  })

  it('renders correctly', () => {
      const component = shallow(<TodoForm />)
      expect(component).toMatchSnapshot()
  })
})
