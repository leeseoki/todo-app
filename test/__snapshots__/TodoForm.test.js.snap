// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="form-control"
          placeholder="Enter new todo"
          type="text"
        />,
        <button
          className="btn btn-primary"
          type="submit"
        >
          Add
        </button>,
      ],
      "className": "form-inline",
      "onSubmit": [Function],
    },
    "ref": "form",
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "placeholder": "Enter new todo",
          "type": "text",
        },
        "ref": "item",
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add",
          "className": "btn btn-primary",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Add",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="form-control"
            placeholder="Enter new todo"
            type="text"
          />,
          <button
            className="btn btn-primary"
            type="submit"
          >
            Add
          </button>,
        ],
        "className": "form-inline",
        "onSubmit": [Function],
      },
      "ref": "form",
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "placeholder": "Enter new todo",
            "type": "text",
          },
          "ref": "item",
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "className": "btn btn-primary",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
